interpreter/Makefile

```diff
@@ -9,131 +9,44 @@
-OCB =		$(OCBA) $(LIBS:%=-libs %)
-JSO =		js_of_ocaml -q --opt 3
-JS =		# set to JS shell command to run JS tests, empty to skip
+JSLIB =		wast.js
 
+BUILDDIR = _build/default
 
-# Main targets
+JS =		# set to JS shell command to run JS tests, empty to skip
 
-.PHONY:		default opt unopt libopt libunopt jslib all land zip smallint dunebuild
 
-default:	opt
-debug:		unopt
-opt:		$(OPT)
-unopt:		$(UNOPT)
-libopt:		_build/$(LIB).cmx _build/$(LIB).cmxa
-libunopt:	_build/$(LIB).cmo _build/$(LIB).cma
-jslib:		$(JSLIB).js
-all:		unopt opt libunopt libopt test
-land:		$(WINMAKE) all
-zip: 		$(ZIP)
-smallint:	smallint.native
-ci:			land jslib dunebuild
+# Main targets
 
-dunebuild:
-	dune build
+.PHONY:		default opt jslib all zip smallint
 
+default:	$(OPT)
+jslib:		$(JSLIB)
+all:		$(OPT) test
+zip:		$(ZIP)
+smallint:	smallint.exe
+ci:		all jslib
 
 # Building executable
+.PHONY:		$(NAME).exe
+$(NAME).exe:
+	rm -f $(NAME)
+	dune build $@
+	cp $(BUILDDIR)/$(OPT) $(NAME)
 
-empty =
-space =		$(empty) $(empty)
-comma =		,
-
-.INTERMEDIATE:	_tags
-_tags:
-		echo >$@ "true: bin_annot"
-		echo >>$@ "true: debug"
-		echo >>$@ "<{$(subst $(space),$(comma),$(DIRS))}/*.cmx>: for-pack($(PACK))"
-
-$(UNOPT):	main.byte
-		mv $< $@
-
-$(OPT):		main.native
-		mv $< $@
-
-.PHONY:		main.byte main.native
-main.byte:	_tags
-		$(OCB) -quiet $@
-
-main.native:	_tags
-		$(OCB) -quiet $@
-
-.PHONY:		smallint.byte smallint.native
-smallint.byte: _tags
-		$(OCB) -quiet $@
-smallint.native: _tags
-		$(OCB) -quiet $@
-
-
-# Building library
-
-FILES =		$(shell ls $(DIRS:%=%/*) | grep '[.]ml[^.]*$$')
-PACK =		$(shell echo `echo $(LIB) | sed 's/^\(.\).*$$/\\1/g' | tr [:lower:] [:upper:]``echo $(LIB) | sed 's/^.\(.*\)$$/\\1/g'`)
-
-.INTERMEDIATE:	$(LIB).mlpack
-$(LIB).mlpack:	$(DIRS)
-		ls $(FILES) \
-		| sed 's:\(.*/\)\{0,1\}\(.*\)\.[^\.]*:\2:' \
-		| grep -v main \
-		| sort | uniq \
-		>$@
-
-.INTERMEDIATE:	$(LIB).mllib
-$(LIB).mllib:
-		echo Wasm >$@
-
-_build/$(LIB).cmo: $(FILES) $(LIB).mlpack _tags Makefile
-		$(OCB) -quiet $(LIB).cmo
-
-_build/$(LIB).cmx: $(FILES) $(LIB).mlpack _tags Makefile
-		$(OCB) -quiet $(LIB).cmx
-
-_build/$(LIB).cma: $(FILES) $(LIB).mllib _tags Makefile
-		$(OCBA) -quiet $(LIB).cma
-
-_build/$(LIB).cmxa: $(FILES) $(LIB).mllib _tags Makefile
-		$(OCBA) -quiet $(LIB).cmxa
-
+.PHONY:		smallint.exe
+smallint.exe:
+	dune build $@
 
 # Building JavaScript library
 
-JSLIB_DIR =		meta/jslib
-JSLIB_FLAGS =	-I $(JSLIB_DIR) -use-ocamlfind -pkg js_of_ocaml -pkg js_of_ocaml-ppx
-
-.INTERMEDIATE:	$(JSLIB).byte
-$(JSLIB).byte:	$(JSLIB_DIR)/$(JSLIB).ml
-		$(OCBA) $(JSLIB_FLAGS) $@
-
-$(JSLIB).js: $(JSLIB).byte
-		$(JSO) $<
-
-# Building Windows build file
-
-$(WINMAKE):	clean
-		echo rem Auto-generated from Makefile! >$@
-		echo set NAME=$(NAME) >>$@
-		echo if \'%1\' neq \'\' set NAME=%1 >>$@
-		$(OCB) main.byte \
-		| grep -v ocamldep \
-		| grep -v mkdir \
-		| sed s:`which ocaml`:ocaml:g \
-		| sed s:main/main.d.byte:%NAME%.exe: \
-		>>$@
-
+$(JSLIB):
+	rm -f $@
```

Comment:
That is odd, why is this necessary? You probably want to make the copy via a target, so that dependency tracking works properly.

Reply:
This should work:
```
$(JSLIB):	$(BUILDDIR)/$(JSLIB)
	cp $< $@

$(BUILDDIR)/$(JSLIB):
	dune build $(JSLIB)
```

Reply:
But what does that have to do with the Makefile? Wouldn't you get that error either way?

Reply:
The Makefile is the one putting `wast.js` at the root of the repository. It works with your fix as long as one doesn't try to run `make wast.js` and then `dune build wast.js`.

I'm not going to use the Makefile anyway so I don't really care. And I believe people using the Makefile are not going to run dune manually so this is probably fine...